component/component.go

```diff
@@ -29,6 +29,11 @@ type Component interface {
 	// Start() then the collector startup will be aborted.
 	// If this is an exporter component it may prepare for exporting
 	// by connecting to the endpoint.
+	//
+	// The context passed to this method must be used only in the scope of Start, and
+	// should not be stored or reused in any async work that the component executes
+	// after the start. It is important because the context may be cancelled by the
```

Comment:
```suggestion
	// after the start. It is important because the context will be cancelled by the caller after Start() returns and startup sequence is completed.
	If the component needs to perform a long-running starting operation then it is recommended
	that Start() returns quickly and the long-running operation is performed in background.
	In that case make sure that the long-running operation does not use the context passed
	to Start() function since that context will be cancelled soon and can abort the long-running operation. Create a new context from the context.Background() for long-running operations. 
```